### PR TITLE
Release 2.1.0

### PR DESCRIPTION
**Pull request by mpgxvii**
- Adds support for aRMT app questionnaire schedule generation (#85)
- Adds support for triggering questionnaires/tasks for a user
- Stores tasks and task state events for the user

### CODE REVIEW

File: src/main/java/org/radarbase/appserver/service/questionnaire/protocol/SimpleReminderHandler.java [hunk 1]
```diff
@@ -45,7 +45,8 @@ public List<Notification> generateReminders(List<Task> tasks, Assessment assessm
                         return notification;
                     });
                         }
-                ).collect(Collectors.toList());
+                )
+                .filter(notification -> (Instant.now().isBefore(notification.getScheduledTime()))).collect(Collectors.toList());
```
Comment (yatharthranjan) [inline src/main/java/org/radarbase/appserver/service/questionnaire/protocol/SimpleReminderHandler.java:49]: Should we still check the ttl or completion window here. the reminder is valid till the task is valid.

File: src/main/java/org/radarbase/appserver/controller/ProtocolEndpoint.java [hunk 2]
```diff
@@ -71,6 +83,10 @@ public Protocol getProtocolUsingProjectIdAndSubjectId(
                   + PathsUtil.PROJECT_ID_CONSTANT
                   + "/"
                   + PathsUtil.PROTOCOL_PATH)
+  @Authorized(
+    permission = AuthPermissions.READ,
+    entity = AuthEntities.SUBJECT,
+    permissionOn = PermissionOn.PROJECT)
```
Comment (yatharthranjan) [inline src/main/java/org/radarbase/appserver/controller/ProtocolEndpoint.java:89]: I think you can just have permission = READ, entity = PROJECT here

**Review by yatharthranjan — COMMENTED** [review]
looks good, some final comments

File: src/main/java/org/radarbase/appserver/controller/ProtocolEndpoint.java [hunk 3]
```diff
@@ -43,6 +47,10 @@ public ProtocolEndpoint(ProtocolGenerator protocolGenerator) {
   }
 
   @GetMapping("/" + PathsUtil.PROTOCOL_PATH)
+  @Authorized(
+      permission = AuthPermissions.READ,
+      entity = AuthEntities.SUBJECT,
+      permissionOn = PermissionOn.PROJECT)
```
Comment (yatharthranjan) [inline src/main/java/org/radarbase/appserver/controller/ProtocolEndpoint.java:53]: This not correct. If authorising on project or subject, you need both as path params in the request

File: src/main/java/org/radarbase/appserver/service/questionnaire/protocol/SimpleReminderHandler.java (hunk 1 above)
Comment (mpgxvii) [reply src/main/java/org/radarbase/appserver/service/questionnaire/protocol/SimpleReminderHandler.java]: Ah yes I updated this on the notification handler but didn't for the reminder handler. I've updated it now.

**Review by yatharthranjan — APPROVED** [review]
LGTM, thanks.

**Pull request merged**